### DISCUSSION TITLE
Langsmith + streaming?

### DISCUSSION THREAD

**Discussion by jstjoe** [langchain-ai/langsmith-sdk#575] [Q&A] [answered]
Hi I'm new. I'm using Langchain JS with streaming enabled and am struggling to log any runs / traces to Langsmith.

I've configured the enviornment variables and the client. I've tried manually setting the env varibles in code by creating a LangChainTracer callback and passing that into my `chain.stream()` call like `chain.stream({ input: messages, callbacks })` where callbacks is an array containing the LangChainTracer.

Help?? I can't find anything in the docs which says I need to do something special for streaming but I suspect that's the issue? Is streaming not supported?

**Comment by hinthornw** [comment, marked as answer]
Callbacks  goes within the config (second argument) rather than the input (first argument) - does that resolve your issue?

**Comment by jstjoe** [comment]
🤦 yes, thank you!
